test/infer/test_mcmc.py

```diff
@@ -897,6 +900,16 @@ def model():
     mcmc.run(random.PRNGKey(0))
 
 
+def test_SA_gradient_free():
```

Comment:
how does this test anything?

Reply:
This is just to make sure that no grad call is performed. Here the model has `while_loop`, which we can't take grad (we need to use jacfwd for this).